src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Documents/TextEditorCopyPaste.cs

```diff
@@ -109,14 +109,14 @@ internal static DataObject _CreateDataObject(TextEditor This, bool isDragDrop)
                 // ConfirmDataFormatSetting rasies a public event - could throw recoverable exception.
                 if (ConfirmDataFormatSetting(This.UiScope, dataObject, DataFormats.Text))
                 {
-                    CriticalSetDataWrapper(dataObject,DataFormats.Text, textString);
+                    ((DataObject)dataObject).SetData(DataFormats.Text, textString, false);
```

Comment:
is this cast necessary?

Reply:
Push a quick update if you think it would help.

Reply:
@ojhad, @stevenbrix, @rladuca, @miguep, there are test failures that we need to figure out just now.... to keep this PR moving along in time for P7.

Reply:
@ojhad let us know if you need any help with them